code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityState.java

```diff
@@ -162,6 +162,97 @@ void removeCustomerIdentifiers(final IdentityMap map) {
 		IdentityStorageService.savePropertiesToPersistence(identityProperties);
 	}
 
+	/**
+	 * Construct the advertising identifier consent request event data using the provided consent value
+	 * @param consentVal the consent value defined by {@link IdentityConstants.XDMKeys.Consent#YES}
+	 *                   or {@link IdentityConstants.XDMKeys.Consent#NO}
+	 * @return the event data for advertising identifier consent request
+	 */
+	private Map<String, Object> buildConsentAdIdRequestData(final String consentVal) {
+		// build the map from the bottom level -> up
+		Map<String, Object> consentValMap = new HashMap<>();
+		consentValMap.put(IdentityConstants.XDMKeys.Consent.VAL, consentVal);
+		consentValMap.put(IdentityConstants.XDMKeys.Consent.ID_TYPE, IdentityConstants.Namespaces.GAID);
+
+		Map<String, Object> adIDMap = new HashMap<>();
+		adIDMap.put(IdentityConstants.XDMKeys.Consent.AD_ID, consentValMap);
+
+		Map<String, Object> consentMap = new HashMap<>();
+		consentMap.put(IdentityConstants.XDMKeys.Consent.CONSENTS, adIDMap);
+		return consentMap;
+	}
+
+	/**
+	 * Dispatches a consent request event with the consent value passed
+	 *
+	 * @param consentVal the consent value to send in the event, from
+	 * {@link IdentityConstants.XDMKeys.Consent#YES}/{@link IdentityConstants.XDMKeys.Consent#NO}
+	 */
+	private void dispatchAdIdConsentRequestEvent(final String consentVal) {
+		Map<String, Object> consentData = buildConsentAdIdRequestData(consentVal);
+
+		final Event consentEvent = new Event.Builder(
+			IdentityConstants.EventNames.CONSENT_UPDATE_REQUEST_AD_ID,
+			IdentityConstants.EventType.EDGE_CONSENT,
+			IdentityConstants.EventSource.UPDATE_CONSENT
+		)
+			.setEventData(consentData)
+			.build();
+
+		// Callback is not required because there is no response for this type of event
+		MobileCore.dispatchEvent(
+			consentEvent,
+			new ExtensionErrorCallback<ExtensionError>() {
+				@Override
+				public void error(ExtensionError extensionError) {
+					MobileCore.log(
+						LoggingMode.DEBUG,
+						LOG_TAG,
+						"Failed to dispatch consent event " +
+						consentEvent.toString() +
+						": " +
+						extensionError.getErrorName()
+					);
+				}
+			}
+		);
+	}
+
+	// update advertising identifier is the main entrypoint for the ad ID changes
+	// it will update persistent storage, share the XDM state, and dispatch consent
+
+	void updateAdvertisingIdentifier(
+		final Event event,
+		IdentityProperties identityProperties,
+		final SharedStateCallback callback
+	) {
+		final String newAdId = EventUtils.getAdID(event);
+		if (identityProperties == null) {
+			identityProperties = new IdentityProperties();
+		}
+		String currentAdId = identityProperties.getAdId();
+		if (currentAdId == null) {
+			currentAdId = "";
+		}
+
+		// The ad ID has changed
+		if (newAdId != null && !newAdId.equalsIgnoreCase(currentAdId)) {
```

Comment:
The format for this one should be consistent, though we can use simple equals here to update the value if the case changes. 
Nit: maybe we can reverse this to:
```
if (currentAdId.equals(newAdId) {
return; // no op
}
// handle adid changed case
...
```

Reply:
Logic updated to suggested if check